### DISCUSSION TITLE
How to organize backups and backups of backups?

### DISCUSSION THREAD

**Discussion by wsw70** [KenKundert/emborg#80] [Q&A] [answered]
I would like to backup a few directories (say - `/etc` and `/var/lib/docker/volumes`) locally and on a remote machine. After reading the documentation I am not sure how I should organize this. Would it be
- one file per target-directory combo, which would mean 2 x 2 = 4 files?
- or a file with the targets, and two files, one for each directory? (so 1 + 2 = 3 files)
- or the other way round (still 3 files)
- or one file with everything: all the directories (`src_dirs = '/etc,/var/lib/docker/volumes'`) and both targets (not sure how to configure that), in one single file?

I used my own Borg script for years but I just discovered emborg and it looks really neat!

**Comment by wsw70** [comment]
It seems that I can get away with a `settings` file common to both backups and just having in each specific files the line 
```
repository = '<a local, then a remote path>'
```

**Comment by KenKundert** [comment]
My recommendation would be to create two configurations, one that is responsible for backing up locally and one that backs up to your remote host, and for both simply list both source directories.  Then, create a composite configuration that allows you to back up to both destinations with one invocation of emborg:

```
settings:
    configurations = "to_local to_remote to_both=to_local,to_remote"
    default_configuration = 'to_both'

to_local:
     src_dirs = '/etc /var/lib/docker/volumes'
     repository='/...'

to_remote:
     src_dirs = '/etc /var/lib/docker/volumes'
     repository='remote:/...'
```

If you want to share the remaining settings you can create a 4th file that is included into your two configurations.

**Reply by wsw70** [reply]
Thank you. ~~I am a bit confused now with what goes where.~~ (I think I am not confused anymore :) - see the edit below)

I thought that there **must** be a file called `settings` that lists the common settings and that there **must** be extra files for backups (a string must for the latter, not only a "may"). In my case everything is common except the target repository (one is local, the other one is remote - these are two separate backups of the same files).

I therefore created: 

- a `settings` file with everything I needed
- and a `srv` file (which means "local backup") which only has the line `repository = '/backup/borg/srv'`

(there is no remote yet, I am testing the minimal files).

It works fine.

~~Now, when I read your comment it seems that **everything** could be squeezed in a single `settings` file since it is possible to have a source+repo structure (the `to_local` and `to_remote` in your example)~~
**EDIT**: I tried that but I think I misunderstood your example: these are not sections (`settings:`, `to_local:`, ...) but three separate files
In this case I will have 3 files, with a one-liner in two of them (`repository=`)

**Comment by KenKundert** [comment, marked as answer]
Yes, exactly.  Three different files.
